qiskit/aqua/components/optimizers/aqgd.py

```diff
@@ -39,150 +41,307 @@ class AQGD(Optimizer):
+            maxiter: Maximum number of iterations (full gradient steps)
             eta: The coefficient of the gradient update. Increasing this value
-                 results in larger step sizes: param = previous_param - eta * deriv
-            tol: The convergence criteria that must be reached before stopping.
-                 Optimization stops when: absolute(loss - previous_loss) < tol
-            disp: Set to True to display convergence messages.
-            momentum: Bias towards the previous gradient momentum in current update.
-                      Must be within the bounds: [0,1)
-
+                results in larger step sizes: param = previous_param - eta * deriv
+            momentum: Bias towards the previous gradient momentum in current
+                update. Must be within the bounds: [0,1)
+            param_tol: Tolerance for change in norm of parameters.
+            objective_tol: Tolerance for change in windowed average of objective values.
+                Convergence occurs when either objective tolerance is met OR parameter
+                tolerance is met.
+            averaging: Length of window over which to average objective values for objective
+                convergence criterion
+
+        Raises:
+            AquaError: If the number of iterations doesn't match momentum or eta for the
+                       desired steps.
         """
-        validate_range_exclusive_max('momentum', momentum, 0, 1)
         super().__init__()
+        if maxiter is None:
+            maxiter = [1000]
+        if eta is None:
+            eta = [1.0]
+        if momentum is None:
+            momentum = [0.5]
+        if len(maxiter) != len(eta) or len(maxiter) != len(momentum):
+            raise AquaError("AQGD input parameter length mismatch")
+        for m in momentum:
+            validate_range_exclusive_max('momentum', m, 0, 1)
 
         self._eta = eta
         self._maxiter = maxiter
-        self._tol = tol if tol is not None else 1e-6
-        self._disp = disp
-        self._momentum_coeff = momentum
-        self._previous_loss = None
+        self._momenta_coeff = momentum
+        self._param_tol = param_tol if param_tol is not None else 1e-6
+        self._objective_tol = objective_tol if objective_tol is not None else 1e-6
+        self._averaging = averaging
+        self._avg_objval = None
+        self._prev_param = None
+
+        # state
+        self._eval_count = 0    # function evaluations
+        self._prev_loss = []    # type: List[float]
+        self._prev_grad = []    # type: List[List[float]]
 
-    def get_support_level(self):
-        """ Return support level dictionary """
+    def get_support_level(self) -> Dict[str, int]:
+        """ Support level dictionary
+
+        Returns:
+            Dict[str, int]: gradient, bounds and initial point
+                            support information that is ignored/required.
+        """
```

Comment:
Good question. I don't know since I'm an expert (yet) in optimizers and the base class does not specify the return type. So, what is the return type?

Reply:
Its a dictionary like this below that returns the support level for gradient, bounds and initial point for the optimizer in the optimize method - ignored for instance meaning you can give it a gradient, it will not raise an error or anything rather it will just ignore it. So its the dict of string keys to optimizer support level values.

Reply:
Clear. Fixed, I hope.